src/partition/inertial_flow.cpp

```diff
@@ -7,16 +20,105 @@ namespace partition
 
 InertialFlow::InertialFlow(const GraphView &view_) : view(view_) {}
 
-std::vector<bool> InertialFlow::ComputePartition(const double balance, const double source_sink_rate)
+std::vector<bool> InertialFlow::ComputePartition(const double balance,
+                                                 const double source_sink_rate)
 {
+    auto cut = bestMinCut(10 /* should be taken from outside */, source_sink_rate);
+
+    // TODO: only dummy impl. below:
+
     std::vector<bool> partition(view.NumberOfNodes());
     std::size_t i = 0;
-    for( auto itr = partition.begin(); itr != partition.end(); ++itr )
+    for (auto itr = partition.begin(); itr != partition.end(); ++itr)
     {
         *itr = (i++ % 2) != 0;
     }
     return partition;
 }
 
+InertialFlow::SpatialOrder InertialFlow::makeSpatialOrder(const double ratio,
+                                                          const double slope) const
+{
+    struct NodeWithCoordinate
+    {
+        NodeWithCoordinate(NodeID nid_, util::Coordinate coordinate_)
+            : nid{nid_}, coordinate{std::move(coordinate_)}
+        {
+        }
+
+        NodeID nid;
+        util::Coordinate coordinate;
+    };
+
+    using Embedding = std::vector<NodeWithCoordinate>;
+
+    Embedding embedding;
+    embedding.reserve(view.NumberOfNodes());
```

Comment:
This is what I was talking about above: with a planet sized graph we store node ids and coordinates here for the spatial order re-ordering to take place and then return the 0.25*n "first" and "last" node ids.